bundle/direct/dresources/registered_model.go

```diff
@@ -0,0 +1,92 @@
+package dresources
+
+import (
+	"context"
+
+	"github.com/databricks/cli/bundle/config/resources"
+	"github.com/databricks/cli/bundle/deployplan"
+	"github.com/databricks/databricks-sdk-go"
+	"github.com/databricks/databricks-sdk-go/service/catalog"
+)
+
+type ResourceRegisteredModel struct {
+	client *databricks.WorkspaceClient
+}
+
+func (*ResourceRegisteredModel) New(client *databricks.WorkspaceClient) *ResourceRegisteredModel {
+	return &ResourceRegisteredModel{
+		client: client,
+	}
+}
+
+func (*ResourceRegisteredModel) PrepareState(input *resources.RegisteredModel) *catalog.CreateRegisteredModelRequest {
+	return &input.CreateRegisteredModelRequest
+}
+
+func (*ResourceRegisteredModel) RemapState(model *catalog.RegisteredModelInfo) *catalog.CreateRegisteredModelRequest {
+	return &catalog.CreateRegisteredModelRequest{
+		CatalogName:     model.CatalogName,
+		Comment:         model.Comment,
+		Name:            model.Name,
+		SchemaName:      model.SchemaName,
+		StorageLocation: model.StorageLocation,
+		ForceSendFields: filterFields[catalog.CreateRegisteredModelRequest](model.ForceSendFields),
+	}
+}
+
+func (r *ResourceRegisteredModel) DoRefresh(ctx context.Context, id string) (*catalog.RegisteredModelInfo, error) {
+	return r.client.RegisteredModels.Get(ctx, catalog.GetRegisteredModelRequest{
+		FullName:        id,
+		IncludeAliases:  false,
+		IncludeBrowse:   false,
+		ForceSendFields: nil,
+	})
+}
+
+func (r *ResourceRegisteredModel) DoCreate(ctx context.Context, config *catalog.CreateRegisteredModelRequest) (string, *catalog.RegisteredModelInfo, error) {
+	response, err := r.client.RegisteredModels.Create(ctx, *config)
+	if err != nil {
+		return "", nil, err
+	}
+
+	return response.FullName, response, nil
+}
+
+func (r *ResourceRegisteredModel) DoUpdate(ctx context.Context, id string, config *catalog.CreateRegisteredModelRequest) (*catalog.RegisteredModelInfo, error) {
+	updateRequest := catalog.UpdateRegisteredModelRequest{
+		FullName:        id,
+		Comment:         config.Comment,
+		ForceSendFields: filterFields[catalog.UpdateRegisteredModelRequest](config.ForceSendFields, "Owner", "NewName"),
+
+		// Owner is not part of the configuration tree
+		Owner: "",
+
+		// Name updates are not supported yet without recreating. Can be added as a follow-up.
```

Comment:
Could you add a comment that TF also recreates on name change (at least that's what I see from the test).